collections/_article/progress-report-web-export-in-4-3.md

```diff
@@ -0,0 +1,281 @@
+his task load.
+
+I'm well used to the Web and its quirks. I began creating websites in
+<abbr title="Extensible Hypertext Markup Language">XHTML</abbr> and (vanilla) JavaScript, back in the days.
+Flash games were my jam! Enough so that I even worked for a video game studio as an ActionScript 3 developer in 2010.
+And I didn't really stop caring about the Web platform ever since.
+
+Since then, Flash may have died, but online games didn't.
+
+# The new situation
+But what I realized at the <abbr title="Game Developers Conference">GDC</abbr> is that we're entering in sort into a
+Golden Age of Web games. Not only websites like [Poki](https://poki.com/) or
+[Crazy Games](https://www.crazygames.com/) are super popular, but big players are starting to integrate Web games into
+their services, such as
+[Discord Activities](https://support.discord.com/hc/en-us/articles/4422142836759-Activities-on-Discord)
+or [YouTube Playables](https://www.youtube.com/playables). And all these entities want developers to create games for
+their platform. And they are all asking how Godot could help create fast-loading "instant games".
+
+# Godot 4.3
+Making games for the Web using Godot can be a challenge, especially on 4.x, and that means serious revisions are needed.
+Godot 4.3 promises to be one of the best recent releases about Web exports. One of the biggest issues relating to this
+has been properly fixed: single-threaded exports.
+
+## Single-threaded Web export
+
+### Betting on the wrong horse
+[`SharedArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer)s
+were supposed to revolutionize the Web. And they did. That API permits to share memory between
+[Web Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers)
+(which are the Web native "threads").
+
+Unfortunately, we live in the same timeline that of [Spectre and Meltdown exploits](https://meltdownattack.com/).
+
+The result of this is that [browsers greatly reduced where and how you can use that API](https://developer.chrome.com/blog/enabling-shared-array-buffer?hl=en).
+Namely, browsers nowadays require websites to be [cross-origin isolated](https://web.dev/coop-coep/). When isolated, it
+does unlock the potential of `SharedArrayBuffer`s, but at what cost?
+
+At the cost of having the capacity to make remote calls to other websites. _Adieu_ game monetization. _Adieu_
+interacting with an external API.
+
+The issue is that during the development of Godot 4, we bet on
+the wrong horse: we went all-in with using threads with `SharedArrayBuffer`s. We minimized the cost of using
+cross-origin isolated websites. Sadly, it made it so that
+[Winterpixel Games](https://www.winterpixel.com/) (makers of [Rocket Bot Royale](https://rocketbotroyale.winterpixel.io/)
+and [Goober Dash](https://gooberdash.winterpixel.io/)) were unfortunately stuck on Godot 3.x, which supports
+running games on the main thread.
```

Comment:
```suggestion
[Winterpixel Games](https://www.winterpixel.com/) (makers of [Rocket Bot Royale](https://rocketbotroyale.winterpixel.io/)
and [Goober Dash](https://gooberdash.winterpixel.io/)) (and many other commercial game studio's like them) were unfortunately stuck on Godot 3.x, which supports
running games on the main thread.
```

Reply:
I changed the sentence entirely, to focus on the experience of exporting games on itch.io and other websites, as suggested by @akien-mga.

> We know been a hurdle for a lot of users trying to publish their jam games on [itch.io](https://itch.io), or for some other web game platforms which don't allow configuring headers for `SharedArrayBuffer` support.